integration-tests/suites/prometheus.go

```diff
@@ -0,0 +1,33 @@
+package suites
+
+import (
+	"net/http"
+
+	"github.com/prometheus/common/expfmt"
+)
+
+type PrometheusTestSuite struct {
+	IntegrationTestSuiteBase
+}
+
+func (s *PrometheusTestSuite) SetupSuite() {
+	s.StartCollector(false, nil)
+}
+
+func (s *PrometheusTestSuite) TearDownSuite() {
+	s.StopCollector()
+}
+
+// Verify prometheus metrics are being exported
+func (s *PrometheusTestSuite) TestPrometheus() {
+	response, err := http.Get("http://localhost:9090/metrics")
+	s.Require().NoError(err)
+	s.Assert().Equal(response.StatusCode, 200)
+
+	// Parse the body to check it is a valid prometheus output
+	body := response.Body
+	defer body.Close()
+	var parser expfmt.TextParser
+	_, err = parser.TextToMetricFamilies(body)
+	s.Require().NoError(err)
```

Comment:
Maybe check that some of the fields are non-zero.

Reply:
The output for this is a bit weird, I'll add a check to verify the resulting map has some keys in it, think that should be enough.

Reply:
This is enough.